File: frontend/services/barangService.js
import api from "./api";

export const getBarang = async () => {
  const response = await api.get("/barang/");
  return response.data;
};

export const createBarang = async (barangData) => {
  const response = await api.post("/barang/", barangData);
  return response.data;
};

export const updateBarang = async (id, itemData) => {
  const response = await api.put(`/barang/${id}`, barangData);
  return response.data;
};

export const deleteBarang = async (id) => {
  const response = await api.delete(`/barang/${id}`);
  return response.data;
};

File: backend/app/services/barang_service.py
from sqlalchemy.orm import Session
from fastapi import HTTPException, status
from app.models.barang import Barang
from app.schemas.barang_schema import BarangCreate, BarangUpdate
from app.repositories.barang_repository import BarangRepository

class BarangService:

    @staticmethod
    def create_barang(db: Session, data: BarangCreate, gudang_id: int):
        return BarangRepository.create(db, data, gudang_id)

    @staticmethod
    def get_barang_by_role(db: Session, role: str, gudang_id: int):
        if role == "super_admin":
            return BarangRepository.get_all(db)
        return BarangRepository.get_all_by_gudang(db, gudang_id)

    @staticmethod
    def update_barang(db: Session, barang: Barang, data: BarangUpdate):
        if data.stok < 0:
            raise HTTPException(
                status_code=status.HTTP_400_BAD_REQUEST,
                detail="STOK TIDAK BISA MINUS!"
            )

        return BarangRepository.update(db, barang, data)

    @staticmethod
    def delete_barang(db: Session, barang: Barang):
        if barang.stok > 0:
            raise HTTPException(
                status_code=status.HTTP_400_BAD_REQUEST,
                detail="STOK HARUS KOSONG SEBELUM BISA DIHAPUS!"
            )

        return BarangRepository.delete(db, barang)
